scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -507,27 +490,74 @@ def init_and_add_platforms(data, board, target, qualifier, aliases):
                     for rev in board.revisions:
                         if rev.name:
                             target = f"{board.name}@{rev.name}/{qual}"
-                            aliases = [target]
+                            alias2target[target] = target
                             if rev.name == board.revision_default:
-                                aliases.append(f"{board.name}/{qual}")
+                                alias2target[f"{board.name}/{qual}"] = target
                             if '/' not in qual and len(board.socs) == 1:
                                 if rev.name == board.revision_default:
-                                    aliases.append(f"{board.name}")
-                                aliases.append(f"{board.name}@{rev.name}")
+                                    alias2target[f"{board.name}"] = target
+                                alias2target[f"{board.name}@{rev.name}"] = target
                         else:
                             target = f"{board.name}/{qual}"
-                            aliases = [target]
+                            alias2target[target] = target
                             if '/' not in qual and len(board.socs) == 1 \
                                     and rev.name == board.revision_default:
-                                aliases.append(f"{board.name}")
+                                alias2target[f"{board.name}"] = target
 
-                        init_and_add_platforms(data, board, target, qual, aliases)
+                        target2board[target] = board
                 else:
                     target = f"{board.name}/{qual}"
-                    aliases = [target]
+                    alias2target[target] = target
                     if '/' not in qual and len(board.socs) == 1:
-                        aliases.append(board.name)
-                    init_and_add_platforms(data, board, target, qual, aliases)
+                        alias2target[board.name] = target
+                    target2board[target] = board
+
+        for board_dir, data in dir2data.items():
```

Comment:
this suppose to be platform related not test plan related, why moves here

Reply:
`TestPlan.add_configurations()` is already platform related, and as of dfc7860ab1f947e37ed09b1860ad5ea90cefb8ef, it also deals with platform YAML contents directly (see `find_twister_data()`), which in and of itself might be a layering violation. Still, I think this precedent justifies moving the `variants` handling here as well.

The purpose is to allow one Platform instance to be assembled from multiple `twister.yaml` files.

Reply:
Thanks for the feedback. I will hold off on further edits for now, until I can be assured that the overall functional change is positively received.

Reply:
Refactored as suggested. I think it looks significantly better now.